### DISCUSSION TITLE
Is it possible to get the additional login return data from the Notion auth provider?

### DISCUSSION THREAD

**Discussion by madebyisaacr** [supabase/supabase#12278] [Questions]
When using Notion as an auth provider, Notion returns a lot of additional information when a user signs in. From this data, `access_token` is available as `supabase.auth.getSession().data.session.provider_token`, but I can't find a way to retrieve any of the other return data.

Is it possible to access any of the data Notion returns in Supabase? Being able to use some of these properties like `bot_id`, `workspace_name`, `workspace_icon` would be very helpful for the app I'm making.

A screenshot from [Notion developer docs](https://developers.notion.com/docs/authorization#step-4-notion-responds-with-an-access_token-and-some-additional-information) showing the return information:

![image](https://user-images.githubusercontent.com/119639439/217698816-8008668a-9f63-4ab7-9099-b9d4a47d315d.png)

**Comment by nathansudo** [comment]
Upvoted this because I've got the same question, just asked about this in the Notion Devs community and Supabase Discord as well trying to find some answers. 

It seems [some of these were added later to the Notion API](https://developers.notion.com/changelog/oauth-token-response-now-includes-more-info-about-the-workspace), wondering if the Login with Notion integration just needs to be updated to support the additional variables?

@madebyisaacr Did you ever find a solution for this?  Thanks!

**Comment by MintedKenny** [comment]
Looks like you can get some of this data by calling the bot user! You can refer to the docs here, but it's a simple API call once you have the access / provider token: https://developers.notion.com/reference/get-self

**Comment by antonlvovych** [comment]
Any updates?

**Comment by nathansudo** [comment]
Still waiting for a solution on this any updates or acknowledgement from Supabase?
